src/Kernel/Process.class.st

```diff
@@ -218,13 +218,11 @@ Process >> doTerminationFromAnotherProcess [
 	It assumes that self is not the active process and the termination is requested from another process.
 	It assumes the terminator process won't resume before the process represented by self is terminated."	
 		
-	[ ] ensure: [ | oldList context terminator |
-		oldList := self suspend.
-		self isTerminated ifTrue: [^self].
+	[ ] ensure: [ | context terminator |
 		self isTerminating ifTrue: [ProcessAlreadyTerminating signal. ^self].
 		terminating := true.
+		context := self suspend handleProcessTerminationOfWaitingContext: (suspendedContext ifNil: [^self]).
```

Comment:
I prefer the original version where #suspend call is split. It is more readable IMHO and it makes debugging step by step much simpler.

Reply:
sure, I just wanted to make a harmless change to force CI run again :) Pushing...